tsl/src/compression/compression_dml.c

```diff
@@ -84,7 +85,20 @@ decompress_batches_for_insert(const ChunkInsertState *cis, TupleTableSlot *slot)
 	CompressionSettings *settings = ts_compression_settings_get(cis->compressed_chunk_table_id);
 	Assert(settings);
 
-	Bitmapset *key_columns = compressed_insert_key_columns(out_rel);
+	bool covering;
+	Bitmapset *key_columns = compressed_insert_key_columns(out_rel, &covering);
+	OnConflictAction on_conflict = ONCONFLICT_UPDATE;
+	/*
+	 *  When no on conflict clause is specified and the index is covering, we can
+	 *  error out before decompressing anything.
+	 *  For ON CONFLICT DO NOTHING with covering index we can skip decompression
```

Comment:
Let's use some other term for this, "covering index" is a different PG thing.

So this means that the compressed scan checks the constraint conditions, and we do not have to recheck the individual tuples, right?

Reply:
no, the compressed scan alone can almost never check the constraint condition as the constraint will include partitioning columns and those are compressed so cant be used to determine conflict in the compressed scan

Reply:
changed it to covered